usersession/agent/rest_api.go

```diff
@@ -375,11 +398,26 @@ func postPendingRefreshNotification(c *Command, r *http.Request) Response {
 	}
 
 	// TODO: this message needs to be crafted better as it's the only thing guaranteed to be delivered.
-	summary := fmt.Sprintf(i18n.G("Update available for %s."), refreshInfo.InstanceName)
 	var urgencyLevel notification.Urgency
-	var body, icon string
+	var body, icon, name string
 	var hints []notification.Hint
 
+	snapname, instanceKey := snap.SplitInstanceName(refreshInfo.InstanceName)
+	// If we have a desktop file of the busy application, use that apps's icon and name, if possible
+	if refreshInfo.BusyAppDesktopEntry != "" {
+		parser := goconfigparser.New()
+		desktopFilePath := filepath.Join(dirs.SnapDesktopFilesDir, refreshInfo.BusyAppDesktopEntry+".desktop")
+		if err := parser.ReadFile(desktopFilePath); err == nil {
+			icon, _ = parser.Get("Desktop Entry", "Icon")
+			name = combineNameAndKey(getLocalizedAppNameFromDesktopFile(parser, snapname), instanceKey)
+		}
+	}
+	if name == "" {
+		name = combineNameAndKey(snapname, instanceKey)
+	}
+
+	summary := fmt.Sprintf(i18n.G("Update available for %s."), name)
```

Comment:
can you add a unit test to check the notification content? something like ` `TestPostPendingRefreshNotificationFewDays` should be sufficient

Reply:
There are already several tests for that... `TestPostPendingRefreshNotificationHappeningNow()`, `TestPostPendingRefreshNotificationFewDays()`, `TestPostPendingRefreshNotificationFewHours()`, `TestPostPendingRefreshNotificationFewMinutes()`... I can add a test for the name itself...

Reply:
Ok, added four tests to check the extra cases where there is an instance key and/or a desktop file with translatable entries.